pkg/scale/scale.go

```diff
@@ -100,6 +84,21 @@ func (s *Scaler) EnsureScale(ctx context.Context, scale uint) error {
 			return err
 		}
 	}
+	// Before doing any scaling, prune any PVCs that are not currently in use.
+	// This only needs to be done when scaling up but the operation is a noop
+	// if there are no PVCs not currently in use.
+	// As of v20.2.0, CRDB nodes may not be recommissioned. To account for
```

Comment:
I don't know where this is coming from, but we can recommission nodes. see https://www.cockroachlabs.com/docs/v21.1/remove-nodes#step-2-recommission-the-decommissioning-nodes

The node won't automatically recommission, there's a step after the node starts before it's allowed to rejoin the cluster.  And if the node was properly decommissioned there's no need to recommission, a fresh PVC is preferred. but If the decommission didn't complete successfully, recommissioning is the best option to make the cluster healthy again.

Reply:
Nodes that have been fully decommissioned may not be recommissioned. Agreed that if the decommissioning process fails for whatever reason, recommissioning is the best option. However, if it succeeds the PVC can't be reused.